Assets/Scripts/GraphService.cs

```diff
@@ -94,39 +105,72 @@ public void RemoveNode(Node node)
     {
       Destroy(node.gameObject);
       Graph.nodes.Remove(node);
-      var edgesToRemove = Graph.edges.Where(e => e.from == node.id || e.to == node.id).ToList();
+      var edgesToRemove = Graph.FindNodeEdges(node);
 
       foreach (var edge in edgesToRemove)
       {
-        RemoveEdge(edge);
+        RemoveAllEdgesBetween(edge.nodeFrom, edge.nodeTo);
       }
     }
 
-    public void RemoveEdge(Edge edge)
-    {
-      Destroy(edge.gameObject);
-      Graph.edges.Remove(edge);
-    }
-
-    public Node FindNodeById(int id) => Graph.nodes.SingleOrDefault(n => n.id == id);
-
-    public void FixEdge(Edge edge)
+    public void RemoveAllEdgesBetween(Node node1, Node node2)
     {
-      var lineRenderer = edge.gameObject.GetComponent<LineRenderer>();
-      var startingNode = FindNodeById(edge.from);
-      var endingNode = FindNodeById(edge.to);
-      lineRenderer.SetPosition(0, startingNode.Position);
-      lineRenderer.SetPosition(1, endingNode.Position);
+      var edges = Graph.FindEdgesByNodes(node1, node2);
+      foreach (var edge in edges)
+      {
+        Destroy(edge.gameObject);
+        Graph.edges.Remove(edge);
+      }
     }
 
     /// <summary>
     ///   Update edges positions based on corresponding nodes
     /// </summary>
     public void FixEdges()
     {
-      foreach (var e in Graph.edges)
+      foreach (var edgeGroup in Graph.GetEdgesGroupedByNodes())
+      {
+        var (nodeFrom, nodeTo) = GetStartAndEndNodes(edgeGroup.Key);
+        FixEdgeGroup(edgeGroup.Value, nodeFrom, nodeTo);
+      }
+    }
+
+    public void FixEdgesOfNode(Node node)
+    {
+      var edgeLists =
+        from edgeGroup in Graph.GetEdgesGroupedByNodes()
+        where edgeGroup.Key.Contains(node)
+        let nodes = GetStartAndEndNodes(edgeGroup.Key)
+        let nodeFrom = nodes.Item1
+        let nodeTo = nodes.Item2
+        select (edgeGroup.Value, nodeFrom, nodeTo);
+      foreach (var (edges, nodeFrom, nodeTo) in edgeLists)
+      {
+        FixEdgeGroup(edges, nodeFrom, nodeTo);
+      }
+    }
+
+    (Node, Node) GetStartAndEndNodes(HashSet<Node> edgeGroupKey)
```

Comment:
Please put names when returning tuples. Better code documentation and allows extra IDE features. Here it would be:
(Node startingNode, Node endingNode).